modules/aws-emr-cluster/main.tf

```
@@ -25,31 +25,42 @@ resource "aws_emr_cluster" "emr-cluster" {
     key_name                          = var.key_pair_name
   }
 
-
-  master_instance_group {
-    name          = var.master_instance_group_name
-    instance_type = var.master_instance_type
-    # NOTE: value must be 1 or 3
-    instance_count = var.master_group_instance_count
-    # Spot Instance definition
-    bid_price = var.master_bid_price
-    ebs_config {
-      size                 = var.master_ebs_size
-      type                 = var.master_ebs_type
-      volumes_per_instance = var.master_ebs_volumes_count
+  master_instance_fleet {
+    instance_type_configs {
+      instance_type = var.master_instance_type
+      ebs_config {
+        size                 = var.master_ebs_size
+        type                 = var.master_ebs_type
+        volumes_per_instance = var.master_ebs_volumes_count
+      }
     }
+    target_on_demand_capacity = 1
```

I think this should be using `var.master_group_instance_count`:
```suggestion
    target_on_demand_capacity = var.master_group_instance_count
```